### PR TITLE
Run existing interpreter tests with the compiler and VM

### PR DESCRIPTION
**Pull request by SupunS**
Work towards https://github.com/onflow/cadence/issues/3769

## Description

This way, it only needs to switch to the new test-utility method.

I only did it for the `interpreter/while_test.go` as it was one of the simplest ones. If this sounds like a good idea, then we can gradually expand this to other tests.

We might need to abstract/expose a few more things as we go, for other complex tests. But the idea would be the same.

______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: interpreter/while_test.go [hunk 1]
```diff
@@ -24,14 +24,15 @@ import (
...
+	"github.com/onflow/cadence/test_utils"
 	. "github.com/onflow/cadence/test_utils/interpreter_utils"
 )
 
 func TestInterpretWhileStatement(t *testing.T) {
 
 	t.Parallel()
 
-	inter := parseCheckAndInterpret(t, `
+	invokable := test_utils.ParseCheckAndPrepare(t, `
```
Comment (SupunS) [inline interpreter/while_test.go:35]: This is all we need to change in the test files: switch to the new API

File: test_utils/test_utils.go [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+	CheckerConfig      *sema.Config
+	HandleCheckerError func(error)
+}
+
+var compile = false
+
+func init() {
+	// TODO: Set the env-variable and re-run the tests.
+	// compile = os.Getenv("COMPILE") == "true"
+}
```
Comment (SupunS) [inline test_utils/test_utils.go:33]: This is disabled by default, for now. I'll open a follow-up PR to enable them

File: test_utils/test_utils.go [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+		vmInstance:  vmInstance,
+		config:      vmConfig,
+		Interpreter: vmConfig.Interpreter(),
+	}
+
+}
+
+// Below helper functions were copied as-is from `misc_test.go`.
+// Idea is to eventually use the below functions everywhere, and remove them from `misc_test.go`,
+// so that the `misc_test.go` would contain only tests.
```
Comment (SupunS) [inline test_utils/test_utils.go:89]: Similar here, I'll open a follow-up PR to refactor all existing tests to use the API in this file (still using the interpreter, not the compiler), and remove them from `misc_test.go`.

I didn't do it here because it will make this PR huge, and also is not related to "using the compiler in existing tests" (i.e: this is an unrelated refactor)

File: interpreter/while_test.go (hunk 1 above)
Comment (turbolent) [reply interpreter/while_test.go:35]: This is great! Maybe use a dot import for `test_utils` like we have for `interpreter_utils`

File: test_utils/interpreter_utils/values.go [hunk 4]
```diff
@@ -49,20 +49,20 @@ func AssertValueSlicesEqual(t testing.TB, inter *interpreter.Interpreter, expect
 	return true
 }
 
-func ValuesAreEqual(inter *interpreter.Interpreter, expected, actual interpreter.Value) bool {
+func ValuesAreEqual(context interpreter.ValueComparisonContext, expected, actual interpreter.Value) bool {
```
Comment (turbolent) [inline test_utils/interpreter_utils/values.go:52]: 👌

File: test_utils/test_utils.go (hunk 2 above)
Comment (turbolent) [reply test_utils/test_utils.go:51]: We can maybe make this a flag, similar to how we have for the smoke tests (see `runSmokeTests`)

**Review by turbolent — APPROVED** [review]
Great idea and great work!

This will allow us to incrementally run more and more existing tests for interpreter and VM, and will save us from having to duplicate tests. Nice 👌

File: test_utils/test_utils.go (hunk 2 above)
Comment (SupunS) [reply test_utils/test_utils.go:51]: For the moment, maybe we can use the flag, but in the makefile, we would have to enable each package separately. This might also makes sense, since we don't want to run all the tests with the compiler (e.g: type checker tests, tools-tests, etc.). Will update and see.

File: interpreter/while_test.go (hunk 1 above)
Comment (SupunS) [reply interpreter/while_test.go]: done in ecdbe8e3a484ed1d2f9f8c5623d2fb17d8f198b5

File: test_utils/test_utils.go (hunk 2 above)
Comment (SupunS) [reply test_utils/test_utils.go]: Added a temporary solution in https://github.com/onflow/cadence/pull/3813/commits/ecdbe8e3a484ed1d2f9f8c5623d2fb17d8f198b5

**Review by turbolent — APPROVED** [review]
Nice! 👏
